src/test/scripts/check_enterprise_merge.sh

```diff
@@ -4,52 +4,55 @@
 set -u
 # exit immediately if a command fails
 set -e
+# list executed commands
+set -x
```

Comment:
Could it be better to not show the executed commands?

Because this means that by default every command will be shown, and we will need to redirect the output to `dev/null` if we don't want to show it. So in the future if someone adds something and forgets to hide the output, it could leak something.

If the default is not to show the command, then if we want to show the commands, we can echo them or enable it at that point with `set +x`. 

My preference would be the latter, because it seems safer since it doesn't have the aforementioned problem. What do you think?

Reply:
In my experience these scripts always fail at some point and debugging them by only looking at the output is very hard in those cases. I think there's only two cases where you don't want to have `set -x` in CI:
1. During sensitive commands, e.g. containing passwords
2. During `echo`, because then the message is shown twice

I've changed the code to handle these two cases in a better way now.

Now a quick example of why it's useful. One thing that confused me about the previous script was that it would show:
```
fatal: 'community' does not appear to be a git repository
fatal: Could not read from remote repository.
```
This was without context and seemed like a bug. With `set -x` it's much clearer that this was no actual problem, just a forgotten `2> /dev/null`.